packages/radio-player/src/radio-player.ts

```diff
@@ -0,0 +1,738 @@
+import MusicZone from './models/music-zone';
+import RadioPlayerConfig from './models/radio-player-config';
+
+@customElement('radio-player')
+export default class RadioPlayer extends LitElement {
+  @property({ type: RadioPlayerConfig }) config: RadioPlayerConfig | undefined = undefined;
+
+  @property({ type: TranscriptConfig }) transcriptConfig: TranscriptConfig | undefined = undefined;
+
+  @property({ type: Number }) currentTime = 0;
+
+  @property({ type: String }) searchTerm = '';
+
+  @property({ type: Boolean }) skipMusicSections = false;
+
+  @property({ type: Number }) private percentComplete = 0;
+
+  @property({ type: Boolean }) private isPlaying = false;
+
+  @property({ type: Number }) private duration = 0;
+
+  @property({ type: Number }) private playbackRate = 1;
+
+  @property({ type: Number }) private volume = 1;
+
+  @property({ type: Boolean }) private shouldShowSearchResultSwitcher = false;
+
+  @property({ type: Boolean }) private shouldShowNoSearchResultMessage = false;
+
+  private musicZones: MusicZone[] = [];
+
+  render(): TemplateResult {
+    return html`
+      ${this.audioElementTemplate}
+      <section role="main">
+        ${this.titleDateTemplate} ${this.collectionLogoTemplate} ${this.playbackControlsTemplate}
+        <div class="waveform-scrubber-container">
+          ${this.waveFormProgressTemplate} ${this.scrubberBarTemplate}
+        </div>
+        ${this.searchSectionTemplate} ${this.transcriptViewTemplate}
+      </section>
+    `;
+  }
+
+  play(): void {
+    /* istanbul ignore else */
+    if (this.audioElement) this.audioElement.play();
+  }
+
+  pause(): void {
+    /* istanbul ignore else */
+    if (this.audioElement) this.audioElement.pause();
+  }
+
+  seekTo(seconds: number): void {
+    /* istanbul ignore else */
+    if (this.audioElement) this.audioElement.seekTo(seconds);
+  }
+
+  private get titleDateTemplate(): TemplateResult {
+    return html`
+      <div class="title-date">
+        <div class="title">
+          ${this.config ? this.config.title : ''}
+        </div>
+
+        <div class="date">
+          ${this.config ? this.config.date : ''}
+        </div>
+      </div>
+    `;
+  }
+
+  private get collectionLogoTemplate(): TemplateResult {
+    return html`
+      <img class="collection-logo" src=${this.logoUrl} />
+    `;
+  }
+
+  private get logoUrl(): string {
+    return this.config ? this.config.logoUrl : '';
+  }
+
+  private get waveFormProgressTemplate(): TemplateResult {
+    return html`
+      <waveform-progress
+        interactive="true"
+        .waveformUrl=${this.waveformUrl}
+        .percentComplete=${this.percentComplete}
+        @valuechange=${this.valueChangedFromScrub}
+      >
+      </waveform-progress>
+    `;
+  }
+
+  private get zonesOfSilence(): ZoneOfSilence[] {
+    /* istanbul ignore else */
+    if (this.duration === 0) {
+      return [];
+    }
+
+    // eslint-disable-next-line max-len
```

Comment:
we should update our max-len rule

Reply:
I'm not totally sure the best way to handle this. There is a framework called `prettier` that auto formats code according to our rules, but it seems like it's having problems keeping this line under our line length limit. It does a pretty good job, but it kept making the next line greater than our length so I ended up just disabling it for this line.